src/Solvers/batched_tridiagonal_solver.jl

```diff
@@ -76,45 +78,115 @@ function solve!(ϕ, solver::BatchedTridiagonalSolver, rhs, args... )
     a, b, c, t, parameters = solver.a, solver.b, solver.c, solver.t, solver.parameters
     grid = solver.grid
 
-    launch!(architecture(solver), grid, :xy,
-            solve_batched_tridiagonal_system_kernel!, ϕ, a, b, c, rhs, t, grid, parameters, Tuple(args))
+    launch!(architecture(solver), grid, launch_config(grid),
```

Comment:
I think it's better to use an explicit tag specified at the time the solver is built (and which can be controlled / toggled  by the user) rather than to hard code the tridiagonal direction based on grid regularity. The first allows more user control whereas the second (the current approach) hard codes an _assumption_, always a bad thing.